### PR TITLE
Navigation1

### PR DESCRIPTION
**Pull request by victorjefferies**
Camera and manual input

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-  render() {
-    return (
-      <View style={styles.container}>
-        <Text>Open up App.js to start working on your app!</Text>
-      </View>
-    );
-  }
-}
 
-const styles = StyleSheet.create({
```
Comment (AlexDresco) [inline App.js]: That's also valid for our code, but where do all the StyleSheet go? 
I see you deleted it here, but do we leave it on the JS files or in the screens? I'm not too sure.

File: components/InputManual.js [hunk 2]
```diff
@@ -1 +1,41 @@
...
+
+         <Button
+          title="Submit number!"
+          onPress = { () => {
+            console.log(this.state.barCodeNumber) ;
+            this.props.navigateToResult();
+          }}
+          />
+
+
```
Comment (AlexDresco) [inline components/InputManual.js]: Extra lines

File: components/InputManual.js [hunk 3]
```diff
@@ -1 +1,41 @@
...
+import { Text, TextInput, View, StyleSheet, Alert, TouchableOpacity, Button } from 'react-native';
+import { createStackNavigator, StackActions, NavigationActions } from 'react-navigation';
+
+class InputManual extends React.Component {
+  state = {
+    barCodeNumber: ""
+  };
+
+  handleBarCode = (text) => {
+    this.setState({barCodeNumber: text})
```
Comment (AlexDresco) [inline components/InputManual.js]: That's great - then it should integrate with Results.js using url = InputManual.state.barCodeNumber 👍

File: components/InputScanner.js [hunk 4]
```diff
@@ -1 +1,58 @@
...
+      hasCameraPermission: status === 'granted',
+    });
+  };
+
+  _handleBarCodeRead = async (data) => {
+    const { code } = await data
+    this.setState({
+      scannedBarCode: data,     
+    })   
+    console.log(this.state.scannedBarCode)
```
Comment (AlexDresco) [inline components/InputScanner.js]: Do we want to keep the console.log for the final code when we merge to master?

File: components/InputScanner.js [hunk 5]
```diff
@@ -1 +1,58 @@
-// hhjg
+import React, { Component } from 'react';
+import { Text, View, StyleSheet, Alert } from 'react-native';
+import { Constants, BarCodeScanner, Permissions } from 'expo';
+
+class InputScanner extends React.Component {
+  state = {
+    hasCameraPermission: null,
+    scannedBarCode: null,
```
Comment (AlexDresco) [inline components/InputScanner.js]: Nice - again that can be access with InputScanner.state.scannedBarCode. 
We'll need to have a way to check whether the code comes from the manual input or scanned input.

File: screens/HomeScreen.js [hunk 6]
```diff
@@ -1 +1,49 @@
-// home
+import React from 'react';
+import { StyleSheet, Text, View, Button, Image } from 'react-native';
+import { createStackNavigator, StackActions, NavigationActions } from 'react-navigation';
+import { ImageBackground } from 'react-native';
+
+
```
Comment (AlexDresco) [inline screens/HomeScreen.js]: Extra lines

File: components/InputScanner.js [hunk 7]
```diff
@@ -1 +1,58 @@
...
+  };
+
+  _handleBarCodeRead = async (data) => {
+    const { code } = await data
+    this.setState({
+      scannedBarCode: data,     
+    })   
+    console.log(this.state.scannedBarCode)
+    this.props.navigateToResult();
+  };
```
Comment (Ruaridhi) [inline components/InputScanner.js]: can you run through it please

File: components/InputScanner.js [hunk 8]
```diff
@@ -1 +1,58 @@
...
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingTop: Constants.statusBarHeight,
+    backgroundColor: '#ecf0f1',
+  }
+});
+
+export default InputScanner;
```
Comment (Ruaridhi) [inline components/InputScanner.js]: Same as previous 'standardise'

File: screens/HomeScreen.js [hunk 9]
```diff
@@ -1 +1,49 @@
...
+      <View style={styles.container}>
+      <ImageBackground
+          source={require('../assets/images/homeBackground.jpg')}
+          style={styles.homeBackground}>
+          <Text>Home Screen</Text>
+      </ImageBackground>
+        <Button
+        title="Scan"
+        onPress={ () => {
+          this.props.navigation.dispatch(StackActions.reset({
```
Comment (Ruaridhi) [inline screens/HomeScreen.js]: can we just go through again what StackActions is please

File: App.js (hunk 1 above)
Comment (AmyDe) [reply App.js]: We should all look into putting style sheets in a separate file for consistency. Let's look into how to do this.

Comment (Din-s1) [reply App.js]: A single stylesheet will be created to simpify and clean the screens also to improve reusability and decoupling.

File: components/InputManual.js (hunk 3 above)
Comment (AmyDe) [reply components/InputManual.js]: Consistency with barcode number stored state?

File: components/InputManual.js (hunk 2 above)
Comment (Din-s1) [reply components/InputManual.js]: You're an extra line XD

**Pull request merged**